pyproject.toml

```diff
@@ -30,16 +30,17 @@ packages = [
 "Documentation" = "https://pandas.pydata.org/pandas-docs/stable"
 
 [tool.poetry.dependencies]
-python = ">=3.9"
+python = ">=3.9 <3.13"
 types-pytz = ">= 2022.1.1"
 numpy = ">=1.25.0"
 
 [tool.poetry.dev-dependencies]
 mypy = "1.5.1"
-pandas = "2.0.3"
+pandas = "2.1.1"
 pyarrow = ">=10.0.1"
 pytest = ">=7.1.2"
-pyright = ">= 1.1.327"
+# pyright 1.1.327 has bug fixed in 1.1.328
+pyright = "1.1.326"
```

Comment:
Which bug is affecting pandas-stubs?

Reply:
Not clear, but 1.1.328 is installed in VS Code pre-release and within VS Code, I get no complaints, but when I install 1.1.327, it doesn't like the use of this pattern in `core/generic.pyi`:
```python
def somefunc(self: NDFrameT, someargs) -> NDFrameT: ...
```
complaining like this:
```
core\generic.pyi:60:15 - error: Type of parameter "self" must be a supertype of its class "NDFrame" (reportGeneralTypeIssues)
```

1.1.328 is in VSCode, but not on pypi yet

Reply:
Would Self fix the issue?

Reply:
If 1.1.328 also fails once released, then I will try that